mlir/test/Dialect/Vector/vector-transfer-permutation-lowering.mlir

```diff
@@ -145,76 +143,109 @@ func.func @xfer_write_non_transposing_permutation_map(
-     %c0 = arith.constant 0 : index
-      vector.transfer_write %arg0, %mem[%c0, %c0, %c0, %c0], %mask {in_bounds = [true, true], permutation_map = affine_map<(d0, d1, d2, d3) -> (d1, d2)>
-} : vector<4x[8]xi16>, memref<1x4x?x1xi16>
+// CHECK:             vector.transfer_write %[[TRANSPOSE_2]], %[[MEM]]{{.*}}, %[[TRANSPOSE_1]] {in_bounds = [true, true, true]} : vector<4x[8]x1xi16>, memref<1x4x?x1xi16>
+func.func @permutation_with_mask_xfer_write_scalable(
+    %vec: vector<4x[8]xi16>,
+    %mem: memref<1x4x?x1xi16>,
+    %mask: vector<4x[8]xi1>){
 
-    return
+  %c0 = arith.constant 0 : index
+  vector.transfer_write %vec, %mem[%c0, %c0, %c0, %c0], %mask {
+    in_bounds = [true, true],
+    permutation_map = affine_map<(d0, d1, d2, d3) -> (d1, d2)>
+  } : vector<4x[8]xi16>, memref<1x4x?x1xi16>
+
+  return
 }
 
 // transfer_write in MaskOp case not supported.
 // CHECK-LABEL: func @masked_permutation_xfer_write_fixed_width
-//  CHECK-SAME:        %[[ARG_0:.*]]: tensor<?x?xf32>,
-//  CHECK-SAME:        %[[ARG_1:.*]]: vector<16xf32>,
-//  CHECK-SAME:        %[[IDX:.*]]: index,
-//  CHECK-SAME:        %[[MASK:.*]]: vector<16xi1>
+//  CHECK-SAME:   %[[DEST:.*]]: tensor<?x?xf32>,
+//  CHECK-SAME:   %[[VEC:.*]]: vector<16xf32>,
+//  CHECK-SAME:   %[[IDX:.*]]: index,
+//  CHECK-SAME:   %[[MASK:.*]]: vector<16xi1>
 //   CHECK-NOT:   vector.transpose
-//       CHECK:   %[[RES:.*]] = vector.mask %[[MASK]] { vector.transfer_write %[[ARG_1]], %[[ARG_0]]{{.*}} vector<16xf32>, tensor<?x?xf32> } : vector<16xi1> -> tensor<?x?xf32>
-func.func @masked_permutation_xfer_write_fixed_width(%t: tensor<?x?xf32>, %val: vector<16xf32>, %idx: index, %mask: vector<16xi1>) -> tensor<?x?xf32> {
-  %r = vector.mask %mask { vector.transfer_write %val, %t[%idx, %idx] {permutation_map = affine_map<(d0, d1) -> (d0)>} : vector<16xf32>, tensor<?x?xf32> } : vector<16xi1> -> tensor<?x?xf32>
+//       CHECK:   %[[RES:.*]] = vector.mask %[[MASK]] { vector.transfer_write %[[VEC]], %[[DEST]]{{.*}} vector<16xf32>, tensor<?x?xf32> } : vector<16xi1> -> tensor<?x?xf32>
+func.func @masked_permutation_xfer_write_fixed_width(
+    %dest: tensor<?x?xf32>,
+    %val: vector<16xf32>,
+    %idx: index,
+    %mask: vector<16xi1>) -> tensor<?x?xf32> {
+
+  %r = vector.mask %mask {
+    vector.transfer_write %val, %dest[%idx, %idx] {
+      permutation_map = affine_map<(d0, d1) -> (d0)>
+    } : vector<16xf32>, tensor<?x?xf32>
+  } : vector<16xi1> -> tensor<?x?xf32>
+
   return %r : tensor<?x?xf32>
 }
 
 // CHECK-LABEL: func.func @masked_permutation_xfer_write_scalable(
-//  CHECK-SAME:        %[[ARG_0:.*]]: vector<4x[8]xi16>,
-//  CHECK-SAME:        %[[ARG_1:.*]]: tensor<?x?x?x?xf32>,
-//  CHECK-SAME:        %[[MASK:.*]]: vector<4x[8]xi1>)
-//  CHECK-SAME:        -> tensor<?x?x?x?xf32> {
+//  CHECK-SAME:   %[[VEC:.*]]: vector<4x[8]xi16>,
+//  CHECK-SAME:   %[[DEST:.*]]: tensor<?x?x?x?xf32>,
```

Comment:
Is `dest` the nomenclature we want to keep for tensor in opposition to mem for memrefs ?